src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -605,10 +605,11 @@ function copy_to_coupled_boundary!(boundary_condition::BoundaryConditionCoupled{
     i_cell_start, i_cell_step = index_to_start_step_2d(indices[1], axes(mesh_other, 1))
     j_cell_start, j_cell_step = index_to_start_step_2d(indices[2], axes(mesh_other, 2))
 
-    i_cell = i_cell_start
-    j_cell = j_cell_start
+    @threaded for i in 1:length(cells)
```

Comment:
```suggestion
    @threaded for i in eachindex(cells)
```
I think we usually prefer `eachindex`.

Reply:
```suggestion
    # we need indices starting at 1 for the handling of `i_cell` etc.
    Base.require_one_based_indexing(cells)
    
    @threaded for i in eachindex(cells)
```

Reply:
This clarifies what we really need here - and gives the compiler additional information to optimize stuff for us (like `eachindex`) even when bounds check is not disabled manually